artemis-core-client/src/main/java/org/apache/activemq/artemis/core/config/federation/FederationStreamConfiguration.java

```diff
@@ -33,7 +35,7 @@ public String getName() {
       return name;
    }
 
-   public FederationUpstreamConfiguration setName(String name) {
+   public FederationStreamConfiguration setName(String name) {
```

Comment:
Can you make the class <T extends FederationStreamConfiguration> and then here, and other returns, return T, this then mans when its returned you get the correct type, which would have any extra setters or getters.

Reply:
@michaelandrepearce - Good point, I made the change and pushed an update to the branch